practices/practice-one/src/scripts/views/student-item.view.js

```diff
@@ -181,18 +179,20 @@ export default class StudentItemView {
     async #handleViewProfile(id) {
         const respone = await this.#controler.getProfile(id);
         const { isError, message, data } = respone;
-        isError
-            ? alert(message)
-            : data
-            ? (() => {
-                  this.#handleActionOverlay('block', 'block');
-                  this.#addDataFormUpdate(data);
-                  this.#addEventIconUpdate();
-                  this.#addEventButtonUpdate(data.id);
-                  this.#addEventButtonDelete(data.id);
-              })()
-            : alert('please wait');
-
+        if (isError) {
+            alert(message);
+            return respone;
+        }
+        const length = Object.keys(data).length;
+        if (length) {
+            this.#handleActionOverlay('block', 'block');
+            this.#addDataFormUpdate(data);
+            this.#addEventIconUpdate();
+            this.#addEventButtonUpdate(data.id);
+            this.#addEventButtonDelete(data.id);
+            return respone;
+        }
+        alert('please wait');
```

Comment:
This not good about UX, this will pause your app. You can add a spinner to show loading state